go-controller/pkg/ovn/obj_retry.go

```diff
@@ -728,6 +772,15 @@ func (oc *Controller) deleteResource(objectsToRetry *retryObjs, obj, cachedObj i
 			extraParameters.portGroupEgressDenyName,
 			obj)
 
+	case factory.EgressFirewallType:
+		var err error
+		egressFirewall := obj.(*egressfirewall.EgressFirewall)
+		if err = oc.deleteEgressFirewall(egressFirewall); err != nil {
+			err = fmt.Errorf("failed to delete egress firewall %s, error: %v", getEgressFirewallNamespacedName(egressFirewall), err)
+		}
+		metrics.UpdateEgressFirewallRuleCount(float64(-len(egressFirewall.Spec.Egress)))
+		metrics.DecrementEgressFirewallCount()
```

Comment:
As @trozet and @martinkennelly said, let's touch the metrics only if `err == nil`

Reply:
So this is the only remaining part to fix (just like you correctly did above for `addResource`), other than that /lgtm

Reply:
Oohh you return if there's an error, so that's fine actually!! :)